src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java

```diff
@@ -473,29 +488,46 @@ private void getNextBatch(String requestId, Request request, boolean firstBatch,
         logger.info("Instance with instanceId {} wasn't in the request.", instanceId);
       }
     });
-    jsonParser.exceptionHandler(throwable -> {
-      logger.error("Error has been occurred at JsonParser while reading data from response. Message:{}", throwable.getMessage(),
+    jsonParser.exceptionHandler(throwable -> responseChecked.future().onSuccess(invalidResponseReceivedAndProcessed -> {
+        if (invalidResponseReceivedAndProcessed) {
```

Comment:
I don't think that it is a code smell. I can add `invalidResponseReceivedAndProcessed.booleanValue()` if you think that this matters.